### DISCUSSION TITLE
[RFC FS-1038 Discussion] Evaluate generalizable values once

### DISCUSSION THREAD

**Discussion by jack-pappas** [fsharp/fslang-design#222] [Language and core library RFC discussions]
Discussion thread for RFC FS-1038 "Evaluate generalizable values once", based on the suggestion from fsharp/fslang-suggestions#602.

**Comment by jack-pappas** [comment]
@dsyme Do you know where in the compiler I can find the code for generating the backing representation for ``let``-bound values decorated with ``[<GeneralizableValue>]``? I poked around for a bit but haven't found it yet.

**Comment by dsyme** [comment]
@jack-pappas It's a non-trivial bit of work. I suppose a good place to start looking is https://github.com/Microsoft/visualfsharp/blob/44fa027b308681a1b78a089e44fa1ab35ff77b41/src/fsharp/IlxGen.fs#L851.

**Comment by dsyme** [comment]
See my comment here: https://github.com/fsharp/fslang-suggestions/issues/602#issuecomment-344937462 (which should really have been added to this RFC discussion, sorry)

**Comment by dsyme** [comment]
I'm going to archive this RFC due to lack of action.

The evaluate-once would be a breaking change, and it's quite possible to implement a static-type-lookup-table today manually, which is more transparent even if more verbose.
